scripts/go-test.sh

```diff
@@ -1,3 +1,3 @@
 #!/bin/bash
 
-go test -race -v ./...; echo $?
+go test -failfast -race -v ./...; echo $?
```

Comment:
Not sure why do we need a go-test script altogether, but ok

for CI makes sense the `failfast` +1

Reply:
> This is good, but some might prefer to learn of all test failures at once instead. Do you think we should let all tests run in the CI? Its unit tests so shouldn't be taking much time.

I think unit test failure should not be even making it to CI :)
And then if you want to find out everything that is broken - `go test -race -v ./...` locally